docs/db_migration_guide.md

```diff
@@ -0,0 +1,244 @@
+This will mitigate the poor Postgres performance on kvdb.
+
+### SQLite kvdb Migration
+
+- [SQLite migration script](https://github.com/lightninglabs/lndinit/blob/main/docs/data-migration.md#using-sqlite-as-the-destination-remote-database)
+- **Advantage**: Maintains good performance while waiting for full relational migration.
+
+---
+
+## Stage 2: Migration from kvdb to Relational Database
+
+This stage unlocks true SQL performance by restructuring data into relational tables. Migration is **per-subsystem** and **incremental**.
+
+### Subsystem Readiness
+
+| Subsystem           | Relational Backend | Migration Script | Status |
+|---------------------|--------------------|------------------|--------|
+| Invoices            | ✅ Available        | ✅              | Available with **v0.19** |
+| Graph               | ✅ Available        | ✅              | Available with **v0.20** |
+| Payments            | 🚧 In Progress      | Planned          | Targeted with **v0.21**|
+| Btcwallet           | 🚧 In Progress      | Planned          | Targeted with **v0.21**|
+| Forwarding History  | ❌ TBD              | ❌ TBD            | Future work |
+
+---
+
+## Known Limitations and Edge Cases
+
+- **Single database engine required**: LND requires a single consistent backend.
+You cannot run invoices in relational mode while payments remain in kvdb *unless*
+both are on the same SQL engine (e.g., SQLite).
+- **Data loss risk**: Always **back up your `data/` directory** before migration.
+- **Downtime required**: Stage 1 requires LND to be offline. Stage 2 is done at startup, requiring a LND restart.
+- **Postgres kvdb performance**: Postgres performance on kvdb is sub-optimal. It is
+recommended to make the stage 2 migration immediately to avoid performance bottlenecks. Certain RPCs like `listpayments` may not perform well on Postgres if the node has a lot of payments data. If your node operation is heavy
+on payments and `listpayments` performance is critical for you, we'd recommend
+not doing any migration and wait till version 0.21.0 is released.
+- **No migration path between SQL backend**: Once migrated to either Postgres or
+SQLite, it is not possible to switch to the other, so choose your target backend carefully.
+---
+
+## Best Practices for Node Operators
+
+### Choosing the Right Path
+
+- **For most users**: Choose SQLite, then migrate. Later, adopt relational backends subsystem-by-subsystem.
+- **Enterprise/Postgres users**: It is recommened to wait to start the migration 
+until **payments relational backend** is ready, then perform **Stage 1 + Stage 2 in quick succession**.
+
+### Timing Your Migration
+
+- Perform migrations during **low-activity periods**.
+- Monitor LND release notes for relational DB support of different subsystems.
+
+### Backup and Validation
+
+1. Stop LND.
+2. Backup entire `~/.lnd/` directory.
```

Comment:
do we actually recommend this? is it not dangerous to have 2 copies of channel.db lying around?  cause what if we later startup pointing to the wrong one than then broadcast invalid states?